micrometer-core/src/test/java/io/micrometer/core/instrument/binder/system/ProcessorMetricsTest.java

```diff
@@ -32,7 +36,38 @@ void cpuMetrics() {
             assertThat(registry.find("system.load.average.1m").gauge())
                 .describedAs("Not present on windows").isNull();
         } else {
-            assertThat(registry.mustFind("system.load.average.1m").gauge().value()).isGreaterThan(0);
+            assertThat(registry.mustFind("system.load.average.1m").gauge().value()).isGreaterThanOrEqualTo(0);
         }
     }
+
+    @Test
+    void hotspotCpuMetrics() {
+        OperatingSystemMXBean osBean = ManagementFactory.getOperatingSystemMXBean();
+        assumeTrue(osBean.getClass().getName().equals("sun.management.OperatingSystemImpl"));
+
+        MeterRegistry registry = new SimpleMeterRegistry();
+        new ProcessorMetrics().bindTo(registry);
+
+        assertThat(registry.mustFind("system.cpu.usage").gauge().value()).isGreaterThanOrEqualTo(0);
+        assertThat(registry.mustFind("process.cpu.usage").gauge().value()).isGreaterThanOrEqualTo(0);
+    }
+
+    @Test
+    void openJ9CpuMetrics() {
+        OperatingSystemMXBean osBean = ManagementFactory.getOperatingSystemMXBean();
+        assumeTrue(osBean.getClass().getName().equals("com.ibm.lang.management.internal.UnixExtendedOperatingSystem"));
```

Comment:
Change to `assertThat(osBean.getClass().getName()).isEqualTo("com.ibm.lang.management.internal.UnixExtendedOperatingSystem");`

Reply:
JUnits assumeTrue() will skip any further processing and ignore the test: http://junit.org/junit5/docs/current/api/org/junit/jupiter/api/Assumptions.html#assumeTrue-boolean-

The idea is to not let the test fail, like with this proposal :)